frontend/packages/topology/src/components/quick-search/QuickSearchController.tsx

```diff
@@ -1,23 +1,68 @@
 import * as React from 'react';
-import { CatalogService } from '@console/dev-console/src/components/catalog/service/CatalogServiceProvider';
+import { useTranslation } from 'react-i18next';
 import { getQueryArgument } from '@console/internal/components/utils';
+import { quickSearch } from './utils/quick-search-utils';
 import QuickSearchButton from './QuickSearchButton';
 import QuickSearchModal from './QuickSearchModal';
+import { QuickSearchData, QuickSearchProviders } from './utils/quick-search-types';
+import { Button } from '@patternfly/react-core';
+import { Link } from 'react-router-dom';
 
-type QuickSearchControllerProps = CatalogService & {
+type QuickSearchControllerProps = {
   namespace: string;
   viewContainer?: HTMLElement;
+  quickSearchProviders: QuickSearchProviders;
+  allItemsLoaded: boolean;
 };
 
 const QuickSearchController: React.FC<QuickSearchControllerProps> = ({
   namespace,
-  searchCatalog,
-  loaded,
+  quickSearchProviders,
   viewContainer,
+  allItemsLoaded,
 }) => {
   const [isQuickSearchActive, setIsQuickSearchActive] = React.useState<boolean>(
     !!getQueryArgument('catalogSearch'),
   );
+  const { t } = useTranslation();
+
+  const searchCatalog = React.useCallback(
+    (searchTerm: string): QuickSearchData => {
+      return quickSearchProviders.reduce(
+        (acc, quickSearchProvider) => {
+          const items = quickSearchProvider.loaded
+            ? quickSearch(quickSearchProvider.items, searchTerm)
+            : [];
+          const itemCount = items.length;
+          const viewAllLink =
+            itemCount > 0
+              ? [
+                  <Button
+                    variant="link"
+                    isInline
+                    tabIndex={-1}
+                    key={quickSearchProvider.catalogType}
+                  >
+                    <Link to={quickSearchProvider.getCatalogURL(searchTerm, namespace)}>
+                      {t(quickSearchProvider.catalogLinkLabel, {
+                        itemCount,
+                      })}
+                    </Link>
+                  </Button>,
+                ]
+              : [];
+          return {
+            filteredItems: [...acc.filteredItems, ...items].sort((item1, item2) =>
+              item1.name.localeCompare(item2.name),
+            ),
+            viewAllLinks: [...acc.viewAllLinks, ...viewAllLink],
+          };
+        },
+        { filteredItems: [], viewAllLinks: [] },
```

Comment:
I don't think the search function should be responsible for generating the view all links but for now I'm not oppose to it. IMO the view rendering the results should use the data and generate the links itself.